README.md

```diff
@@ -7,290 +7,505 @@
+### Example - Search Users by Email
+
+This endpoint is documented [here](https://auth0.com/docs/api/management/v2#!/Users/get_users).
+
+```php
+$results = $mgmt_api->users->search([
+    'q' => 'josh'
+]);
 
+if (! empty($results)) {
+    echo '<h2>User Search</h2>';
+    foreach ($results as $datum) {
+        printf(
+            '<p><strong>%s</strong> &lt;%s&gt; - %s</p>',
+            !empty($datum['nickname']) ? $datum['nickname'] : 'No nickname',
+            !empty($datum['email']) ? $datum['email'] : 'No email',
+            $datum['user_id']
+        );
+    }
+}
 ```
-$ composer install
-$ php -S localhost:3000
+
+### Example - Get All Clients
+
+This endpoint is documented [here](https://auth0.com/docs/api/management/v2#!/Clients/get_clients).
+
+```php
+$results = $mgmt_api->clients->getAll();
+
+if (! empty($results)) {
+    echo '<h2>Get All Clients</h2>';
+    foreach ($results as $datum) {
+        printf(
+            '<p><strong>%s</strong> - %s</p>',
+            $datum['name'],
+            $datum['client_id']
+        );
+    }
+}
 ```
 
-## Migration guide
+## Contributing
 
-### from 1.x
+We provide and maintain SDKs for the benefit of our developer community. Feedback, detailed bug reports, and focused PRs are appreciated. Thank you in advance!
 
-1. If you use Guzzle (or some other dependency does), you will need to update it to work with Guzzle 6.1.
+When contributing to this SDK, please:
 
-### from 0.6.6
+- Maintain the minimum PHP version (found under `require.php` in `composer.json`).
+- Code to the [PSR-2 standard](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md).
+- Write tests and run them with `composer test`.
+- Keep PRs focused and change the minimum number of lines to achieve your goal.
 
-1. First is important to read the [API v2 changes document](https://auth0.com/docs/apiv2Changes) to catch up the latest changes to the API.
-2. Update your composer.json file.
- - change the version "auth0/auth0-php": "~1.0"
- - add the new dependency "firebase/php-jwt" : "dev-master"
-3. Now the SDK is PSR-4 compliant so you will need to change the namespaces (sorry **:(** ) to `\Auth0\SDK`
-4. The method `getUserInfo` is deprecated and candidate to be removed on the next release. User `getUser` instead. `getUser` returns an User object compliant with API v2 which is a `stdClass` (check the schema [here](https://auth0.com/docs/apiv2#!/users/get_users_by_id))
+To run tests on the SDK, you'll need to create a `.env` file in the root of this package with the following entries:
 
-## Develop
+- `DOMAIN` - Auth0 domain for your test tenant
+- `APP_CLIENT_ID` - Client ID for the test Application
+- `APP_CLIENT_SECRET` - Client Secret for the test Application
+- `GLOBAL_CLIENT_ID` - Client ID for your tenant (found in Tenant > Settings > Advanced)
+- `GLOBAL_CLIENT_SECRET` - Client Secret for your tenant (found in Tenant > Settings > Advanced)
```

Comment:
Just below this I'd add "note that the .env file is added to the .gitignore file so it's not commited." or something like that :D